### ISSUE TITLE
Search for given Date

### ISSUE THREAD

**Issue by saggi-dw** [cosmocode/dokuwiki-plugin-struct#497]
Hi,

it's not an issue :)
I was looking for a way to search for a specific or dynamic date.
For example to search for all events from the current day one year ago.

For this purpose I have extended the SearchConfig.php with the following:
     
```
 //apply date formula, given as strtotime
        if(preg_match('/^(.*?)(?:(\$DATE\((.*?)\)\$?))(.*?)$/', $filter, $match)) {
                $toparse = $match[3];
                if ($toparse == '') {
                    $toparse = 'now';
                }
                if (($timestamp = strtotime($toparse)) === false) {
                        echo "Error in DATE-Filter: ($toparse) nothing to parse.";
                } else {
                    $filter = str_replace($filter,date('Y-m-d',$timestamp),$filter);
                }
        }
```
The new Parameter DATE() use php strtotime values.

This enables a dynamic search for a date:
```
---- struct table ----
schema: tabledata
cols: colname, coldate
filter:  coldate <= $DATE(-1 year)$
filterand: colpage = $ID$
csv: 0
----
```
I am not a programmer, but it would be nice if you could add this. I can write documentation for the feature if needed.

saggi

**Comment by bvdbos** [comment]
@saggi-dw Does your code work with other date-formats (like d-m-Y)?

**Comment by saggi-dw** [comment]
@bvdbos Hi, it should ([https://strtotime.co.uk/?str=17-02-2021](https://strtotime.co.uk/?str=17-02-2021)). But why? You can search for a fixed date by using
`filter:  coldate <= 2020-02-17`
The strtotime parses also dynamic dates.

**Comment by Chris75forumname** [comment]
Hi, I am waiting for this feature to be available, hopefully soon. When will be the next release day with this commit in place? BTW, will it work with something like this: `$DATE(-2 years December 31st)$`, with or without parenthesis?

**Comment by saggi-dw** [comment]
This is the correct syntax: `$DATE(-2 years December 31st)$` and will work. Tested here with `filter: start > $DATE(-2 years December 31st)$`
Basic: `$DATE()$` = now, parentheses are always mandatory. 
https://strtotime.co.uk/?str=-2+years+December+31st

**Comment by annda** [comment]
Closed via c1bfd816f493f6882c3b243ee6fead6d0d406838
